### ISSUE TITLE
Input problem of sam2

### ISSUE THREAD

**Issue by ymax1** [Milad-Khanchi/DepthMOT#1]
Hello, I got this error message. It shows that the format of video_path is incorrect. I found that the input format is a list, which does not match the requirements of sam2. How can I modify it? I look forward to your answer. Thank you.

<img width="1901" height="647" alt="Image" src="https://github.com/user-attachments/assets/c35c9036-789f-4b38-82b9-14425291844a" />

**Comment by Milad-Khanchi** [comment]
Go to this path: `sam2/sam2/utils/misc.py`

change `load_video_frames` function to:

` 

def load_video_frames(
    video_path,
    image_size,
    offload_video_to_cpu,
    img_mean=(0.485, 0.456, 0.406),
    img_std=(0.229, 0.224, 0.225),
    async_loading_frames=False,
    compute_device=torch.device("cuda"),
):


    # if isinstance(video_path, str) and os.path.isdir(video_path):
    #     jpg_folder = video_path
    # else:
    #     raise NotImplementedError(
    #         "Only JPEG frames are supported at this moment. For video files, you may use "
    #         "ffmpeg (https://ffmpeg.org/) to extract frames into a folder of JPEG files, such as \n"
    #         "```\n"
    #         "ffmpeg -i <your_video>.mp4 -q:v 2 -start_number 0 <output_dir>/'%05d.jpg'\n"
    #         "```\n"
    #         "where `-q:v` generates high-quality JPEG frames and `-start_number 0` asks "
    #         "ffmpeg to start the JPEG file from 00000.jpg."
    #     )

    # frame_names = [
    #     p
    #     for p in os.listdir(jpg_folder)
    #     if os.path.splitext(p)[-1] in [".jpg", ".jpeg", ".JPG", ".JPEG"]
    # ]
    # frame_names.sort(key=lambda p: int(os.path.splitext(p)[0]))
    frame_names = video_path
    num_frames = len(frame_names)
    # if num_frames == 0:
    #     raise RuntimeError(f"no images found in {jpg_folder}")
    img_paths = video_path
    img_mean = torch.tensor(img_mean, dtype=torch.float32)[:, None, None]
    img_std = torch.tensor(img_std, dtype=torch.float32)[:, None, None]

    if async_loading_frames:
        lazy_images = AsyncVideoFrameLoader(
            img_paths,
            image_size,
            offload_video_to_cpu,
            img_mean,
            img_std,
            compute_device,
        )
        return lazy_images, lazy_images.video_height, lazy_images.video_width

    images = torch.zeros(num_frames, 3, image_size, image_size, dtype=torch.float32)
    for n, img_path in enumerate(tqdm(img_paths, desc="frame loading (JPEG)")):
        images[n], video_height, video_width = _load_img_as_tensor(img_path, image_size)
    if not offload_video_to_cpu:
        images = images.to(compute_device)
        img_mean = img_mean.to(compute_device)
        img_std = img_std.to(compute_device)
    # normalize by mean and std
    images -= img_mean
    images /= img_std
    return images, video_height, video_width
`

**Comment by ymax1** [comment]
thank you very much
